Automattic/jetpack
# My Jetpack: Add VideoPress stats selector

Fixes #29753.

## Proposed changes:

* Add Redux machinery to fetch stats for products
* Add stats fetching implementation for VideoPress

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to `Jetpack > My Jetpack` and open your browser console
* Run the following line:

```
wp.data.select( 'my-jetpack' ).getProductStats( 'videopress' );
```

* This line will try to get the product stats for VideoPress; it will return `null` the first time because the data is not already set on state, and will trigger the resolver to fetch the data:

<img width="461" alt="Screen Shot 2023-03-30 at 12 30 16" src="https://user-images.githubusercontent.com/6760046/228887717-e2c7dfc2-064f-4a33-889f-3acba853d4eb.png">

* Check on the network tab that a request was made to `/videopress/v1/stats/featured`:

<img width="672" alt="Screen Shot 2023-03-30 at 12 30 49" src="https://user-images.githubusercontent.com/6760046/228887769-3af514f9-fc3c-4afe-b6b7-accbe7236942.png">

* Run the selector again; now it will return the value fetched from the API:

<img width="565" alt="Screen Shot 2023-03-30 at 12 31 14" src="https://user-images.githubusercontent.com/6760046/228888079-c4588cf7-6cb4-41a7-9758-7788fb91172f.png">

* Confirm the lifecyle of the request and also the data on the state using the Redux dev tools:

<img width="1031" alt="Screen Shot 2023-03-30 at 12 37 33" src="https://user-images.githubusercontent.com/6760046/228889767-3ad0a77f-4c6e-4df6-8ffa-11443519a493.png">

* Test using the standalone VideoPress plugin as well as using the Jetpack plugin
